lib/Doctrine/DBAL/Schema/SchemaDiff.php

```diff
@@ -165,4 +169,37 @@ protected function _toSql(AbstractPlatform $platform, $saveMode = false)
 
         return $sql;
     }
+
+    /**
+     * Sorts tables by dependencies so that they are created in the right order.
+     *
+     * This is ncessary when one table depends on another while creating foreign key
+     * constraints directly during CREATE TABLE.
+     *
+     * @return array<\Doctrine\DBAL\Schema\Table>
+     */
+    private function getNewTablesSortedByDependencies()
+    {
+        $commitOrderCalculator = new CommitOrderCalculator();
+        $newTables             = [];
+
+        foreach ($this->newTables as $table) {
+            $newTables[$table->getName()] = true;
+            $commitOrderCalculator->addNode($table->getName(), $table);
+        }
+
+        foreach ($this->newTables as $table) {
+            foreach ($table->getForeignKeys() as $foreignKey) {
+                $foreignTableName = $foreignKey->getForeignTableName();
+
+                if (! isset($newTables[$foreignTableName])) {
+                    continue;
+                }
+
+                $commitOrderCalculator->addDependency($foreignTableName, $table->getName(), 1);
```

Comment:
Currently, this is the only used weight value. Would it make sense to omit it from the implementation or it would be needed for sorting internally anyways?

Reply:
No, the ORM uses it with weights, we can remove he ORM implementation once its here in DBAL.